### PR TITLE
Added stream wrapper support to FileLocator

### PR DESCRIPTION
**Pull request by Bittarman**
This pr allows the FileLocator to support stream wrappers.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -22,7 +22,8 @@
         "php":                       "7.4.*",
         "laminas/laminas-code":      "^3.4.1",
         "ocramius/package-versions": "^1.5.1",
-        "webimpress/safe-writer":    "^2.0"
+        "webimpress/safe-writer":    "^2.0",
+        "mikey179/vfsstream": "^1.6"
```
Comment (Ocramius) [inline composer.json]: Should be a Dev dependency

File: src/ProxyManager/FileLocator/FileLocator.php [hunk 2]
```diff
@@ -24,10 +25,12 @@ public function __construct(string $proxiesDirectory)
         $absolutePath = realpath($proxiesDirectory);
 
         if ($absolutePath === false) {
-            throw InvalidProxyDirectoryException::proxyDirectoryNotFound($proxiesDirectory);
+            if (! is_dir($proxiesDirectory)) {
```
Comment (Ocramius) [inline src/ProxyManager/FileLocator/FileLocator.php]: Hmm, this is an added check: wasn't the problem the realpath call?

Comment (Bittarman) [reply src/ProxyManager/FileLocator/FileLocator.php]: Yes, but as you mentioned you also needed it to deal with symlinks, this seemed to be the better option, and would result in less change of behaviour.
is_dir will work with a stream wrapper, while realpath will not.

File: composer.json (hunk 1 above)
Comment (Bittarman) [reply composer.json]: oops, sorry. fixed.

File: src/ProxyManager/FileLocator/FileLocator.php (hunk 2 above)
Comment (Ocramius) [reply src/ProxyManager/FileLocator/FileLocator.php]: Discussed in chat: I didn't see (during review) that `$proxiesDirectory` is no longer discarded right away

**Comment by Ocramius** [comment]
Thanks @Bittarman!

**Pull request merged**